pkg/asset/agent/image/ignition.go

```diff
@@ -323,6 +328,13 @@ func addHostConfig(config *igntypes.Config, agentConfig *agentconfig.Asset) erro
 	return nil
 }
 
+func addExtraManifests(config *igntypes.Config, extraManifests *manifests.ExtraManifests) {
```

Comment:
Would be great to unit test this function.

Reply:
Indeed, the PR is still in WIP, I'm currently working on adding the unit tests (but this would give you a chance to look at the PR before it's ready)

Reply:
Note: I've added a more generic `TestIgnition_Generate` so that we could test properly the asset class just from the public methods (and avoid using the private ones), providing some support so that it could be easily extended in future also to cover the other cases